### ISSUE TITLE
Launching Android player jams Kodi

### ISSUE THREAD

**Issue by Zuikkis** [Sandmann79/xbmc#595]
**Addon used**
- [X] Amazon VOD (plugin.video.amazon-test)
- [ ] Browser Launcher (plugin.program.browser.launcher)

- Addon version: plugin.video.amazon-test v0.9.4+matrix.1

**Account type**
- [X] primevideo.com
- [ ] amazon.(com/co.uk/de/jp)

**System Setup (please provide the following information):**
- Hardware: NVidia Shield Pro 2019
- OS version: Android 9
- Kodi version number: 19.3

**Upload Logs**

**Describe the bug**

Playback works fine with IS.A, but obviously only in 1080p. To view 4K/hdr I tried to use the "Android" player. However that "jams" entire Kodi. Screen flashes briefly (under a second) and then GUI is stuck showing the episode listing, nothing happens. I have to force close Kodi and restart.

Except that it actually worked a few times, Primevideo started and played my video. It worked several times during that session. But later I tried again and it failed again. I have tried rebooting Shield and it still doesn't work.

Here's a log snippet, this is with verbose mode on:
```

2021-11-10 23:24:14.316 T:9758     INFO <general>: [Amazon VOD] Manufacturer: b'nvidia'
2021-11-10 23:24:14.317 T:9758     INFO <general>: [Amazon VOD] Starting App: com.amazon.amazonvideo.livingroom Video: https://app.primevideo.com/watch?gti=amzn1.dv.gti.7a1b91e4-dc7a-4437-af84-e66937991dd6
2021-11-10 23:24:14.335 T:9635     INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.amazon-test/?mode=PlayVideo&name=0SHGKA0J8D4G01ZGD647627NEJ&asin=amzn1.dv.gti.7a1b91e4-dc7a-4437-af84-e66937991dd6
2021-11-10 23:24:14.337 T:9885     INFO <general>: Creating InputStream
2021-11-10 23:24:14.349 T:9885     INFO <general>: Creating Demuxer
2021-11-10 23:24:14.358 T:9885     INFO <general>: Opening stream: 0 source: 256
2021-11-10 23:24:14.359 T:9885     INFO <general>: [WHITELIST] Searching the whitelist for: width: 640, height: 352, fps: 25.000, 3D: false
2021-11-10 23:24:14.361 T:9885     INFO <general>: Display resolution ADJUST : 3840x2160 @ 50.000000 - Full Screen (36) (weight: 0.000)
2021-11-10 23:24:14.453 T:9635     INFO <general>: CVideoPlayer::CloseFile()
2021-11-10 23:24:14.457 T:9635     INFO <general>: VideoPlayer: waiting for threads to exit
2021-11-10 23:25:10.811 T:9692    ERROR <general>: SOCK: Error selecting socket(s)
2021-11-10 23:25:10.845 T:9692    ERROR <general>: ES: Exception caught while listening for socket
2021-11-10 23:25:10.845 T:9692     INFO <general>: ES: UDP Event server stopped

```

I think the SOCK error is me killing Kodi? It's almost a minute later after the VideoPlayer messages.

I guess the VideoPlayer thing is what briefly flashes on screen before Kodi halts? It's showing some short dummy video to make framerate switch or something?

**Comment by Zuikkis** [comment]
Just after issuing this, I noticed you have fixed this in later (un-released) version:
https://github.com/Sandmann79/xbmc/commit/e2201dc3caa0cc7c28c18d8ccc98f6f59b1192ac#diff-557f6ac862fb919eaeedde10dcb67dddbb5343ade315bd58e098f31570a712fe

But what's the point to play a wav? I thought the whole point of dummy video would be the framerate switching, which Primevideo app doesn't do by itself.

**Comment by Varstahl** [comment]
The dummy is just used to bootstrap the player, allow controls and subclassing iirc. It could be anything. By being an audio file, instead of video, you don't have framerate switching problems because the rate is not changed to match the video because there isn't one.

Also I'm a bit behind the development of the addon and haven't sync with the other dev, so I might also be missing a part of the answer.

**Comment by Sandmann79** [comment]
With enabled framerate switching Kodi crashes while launching Prime Video app.

The dummy is used to get rid of Kodis "One or more items can not be played" dialog, which is displayed when Kodi expects a video, but get's none (e.g. at playback via browser or app)

**Comment by Sandmann79** [comment]
Fixed with 69da66d2949ae6bf3b3e63c33ffe0ff4012de242
